transformer_engine/common/include/transformer_engine/userbuffers.h

```diff
@@ -0,0 +1,260 @@
+/*************************************************************************
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See LICENSE for license information.
+ ************************************************************************/
+
+#include <cuda.h>
+#include <mpi.h>
+#include "cuda_runtime.h"
+
+#ifdef MULTINODE
+#ifndef NOSHARP
+#include <api/sharp.h>
+#endif
+#include <pthread.h>
+#include <chrono>
+#include "gdrapi.h"
+#endif
+
+#define MAX_REGIONS 16
```

Comment:
This is public API header, we should make sure that we define only required macros here and the names should be prepended with NVTE_ to avoid name clashes